core/src/main/resources/defaults/blueprints/7.2.17/cdp-sdx-medium-ha-scalable.bp

```diff
@@ -0,0 +1,516 @@
+{
+  "tags":{
+    "shared_services_ready":true
+  },
+  "description":"7.2.17 - Medium SDX Scalable template with Atlas, HMS, Ranger and other services they are dependent on.  Services like HDFS, HBASE, RANGER, HMS have HA",
+  "blueprint":{
+    "cdhVersion":"7.2.17",
+    "displayName":"medium-ha-datalake-scalable",
+    "hostTemplates":[
+      {
+        "cardinality":2,
+        "refName":"master",
+        "roleConfigGroupsRefNames":[
+          "zookeeper-SERVER-BASE",
+          "hbase-MASTER-BASE",
+          "ranger-RANGER_ADMIN-BASE",
+          "hdfs-GATEWAY-BASE",
+          "hive-GATEWAY-BASE",
+          "hbase-GATEWAY-BASE",
+          "solr-GATEWAY-BASE",
+          "kafka-GATEWAY-BASE",
+          "hdfs-NAMENODE-BASE",
+          "hdfs-FAILOVERCONTROLLER-BASE",
+          "hdfs-JOURNALNODE-BASE"
+        ]
+      },
+      {
+        "cardinality":2,
+        "refName":"gateway",
+        "roleConfigGroupsRefNames":[
+          "atlas-ATLAS_SERVER-BASE",
+          "ranger-RANGER_ADMIN-BASE",
+          "hive-HIVEMETASTORE-BASE",
+          "knox-KNOX_GATEWAY-BASE",
+          "solr-GATEWAY-BASE",
+          "hdfs-GATEWAY-BASE",
+          "kafka-GATEWAY-BASE",
+          "hive-GATEWAY-BASE",
+          "hbase-GATEWAY-BASE"
+        ]
+      },
+      {
+        "cardinality":3,
+        "refName":"core",
+        "roleConfigGroupsRefNames":[
+          "hdfs-DATANODE-BASE",
+          "kafka-KAFKA_BROKER-BASE",
+          "solr-SOLR_SERVER-BASE",
+          "hbase-REGIONSERVER-BASE",
+          "solr-GATEWAY-BASE",
+          "hdfs-GATEWAY-BASE",
+          "kafka-GATEWAY-BASE",
+          "hive-GATEWAY-BASE",
+          "hbase-GATEWAY-BASE"
+        ]
+      },
+      {
+        "cardinality":1,
+        "refName":"auxiliary",
+        "roleConfigGroupsRefNames":[
+          "ranger-RANGER_USERSYNC-BASE",
+          "ranger-RANGER_TAGSYNC-BASE",
+          "zookeeper-SERVER-BASE",
+          "hdfs-JOURNALNODE-BASE"
+        ]
+      },
+      {
+        "cardinality":2,
+        "refName":"idbroker",
+        "roleConfigGroupsRefNames":[
+          "knox-IDBROKER-BASE"
+        ]
+      },
+      {
+        "cardinality":0,
+        "refName":"solrHG",
+        "roleConfigGroupsRefNames":[
+          "solr-SOLR_SERVER-BASE"
+        ]
+      },
+      {
+        "cardinality":0,
+        "refName":"storageHG",
+        "roleConfigGroupsRefNames":[
+          "hbase-REGIONSERVER-BASE",
```

Comment:
All of these roles on the extra horizontally scalable hostgroups should be configured as different roles. Config parameters, otherwise, cannot be changed independently for these roles - and the same configs that would apply to the main instance of the role would apply to the scalable ones, and vice versa.

e.g hbase-REGIONSERVER-HS  would be a new role, which would be almost identical to hbase-REGIONSERVER-BASE - except with any custom memory, etc tuning that we may want to do based on this instance size.

Reply:
CM also has the option to set a config across roles - which can be used if a configuration needs to be changed across all roles.
These independent scalable roles should be controllable independently.

Reply:
@sidseth  Why do we need to different configurations for the instances running on the extra horizontally scalable host groups?
Any examples?

Reply:
One simple example: 
The RAZ hostGroup is sized to be larger, and accordingly we tune #threads or memory differently for these instances.

Similarly for the other services like Solr or HBase.

CM allows configs to be set at the service level - which can be used if everything is the same.
If however, we want the flexibility of changing configs only for the new host groups - that can be easily achieved in CM via hostGroup/role level configs.

A single role spanning hostGroups will always tie the configs together.

Reply:
Make sense 
We will handle it as an improvement.  I don't want to hold this PR. We would like to get this shape merged so we can have functional and non-functional testing.